### PR TITLE
fix(wallet): adds utxo selection to manual spend on wallet interface

### LINKED ISSUE #54: Spend specific UTXO rather than all at once
## I'm submitting a…

* [ ] Regression (a behavior that used to work and stopped working in a new release)
* [ ] Bug report 
* [x] Feature request
* [ ] Documentation issue or request
* [ ] Support request

## Expected Behavior
If user has UTXOs with 0.1, 0.7, 0.4, 0.5 BTC each - user should be able to spend 0.1 UTXO without having to spend others back as "change" to self-address

## Current Behavior
Unless I'm misunderstanding, currently Caravan requires spending all UTXOs in one txn.

### PR DESCRIPTION
**Pull request by humanumbrella**
## Description

Allow UTXO selection on manual spend in Wallet interface.

## Does this PR introduce a breaking change?

* [ ] Yes
* [X] No

## Does this PR fix an open issue?

* [X] Yes
* [ ] No

Fixes #54 

Follow-up to https://github.com/unchained-capital/caravan/pull/221 - adds in the necessary logic to handle utxo selection across multiple addresses

### CODE REVIEW

**Comment by bucko13** [comment]
Not sure how we want to handle since this isn't really blocking, but I found a small bug. 

To reproduce: 
1. author a transaction. in my example I have one that's a partial spend from address and a full spend
2. Go to send 
3. Go back to edit
4. Notice the select boxes for the address are no longer selected

What could be blocking though is that I selected only 1/2 of the UTXOs from an address however when going back it looks like both are selected. If this is the case and can be confirmed, I'd say this is blocking since we don't want someone accidentally spending all the UTXOs when they only meant to spend part. 

Screen Shot 2021-06-09 at 11 53 53 AM

**Review by bucko13 — CHANGES_REQUESTED** [review]
Couple of small comments on the code as is, but I think we should address the issue when you go back to edit after clicking to sign

**Comment by waldenraines** [comment]
> Not sure how we want to handle since this isn't really blocking, but I found a small bug.
> 
> To reproduce:
> 
> 1. author a transaction. in my example I have one that's a partial spend from address and a full spend
> 2. Go to send
> 3. Go back to edit
> 4. Notice the select boxes for the address are no longer selected
> 
> What could be blocking though is that I selected only 1/2 of the UTXOs from an address however when going back it looks like both are selected. If this is the case and can be confirmed, I'd say this is blocking since we don't want someone accidentally spending all the UTXOs when they only meant to spend part.

This is also the case if you navigate away from the spend tab (by going to the addresses or receive tab, for example).  When you come back you end up in the same state as shown in the screenshot.

File: package.json [hunk 1]
```diff
@@ -120,6 +120,6 @@
     "redux-thunk": "^2.3.0",
     "reselect": "^4.0.0",
     "unchained-bitcoin": "^0.1.6",
-    "unchained-wallets": "^0.1.20"
+    "unchained-wallets": "^0.1.21"
```
[the remark this answers is not in the document]
Comment (humanumbrella) [reply package.json:123]: Yes - just bumping it to the latest.

File: src/components/ScriptExplorer/UTXOSet.jsx [hunk 2]
```diff
@@ -32,42 +32,95 @@ class UTXOSet extends React.Component {
...
-          checked: true,
+          checked: props.selectAll,
         };
       }),
       toggleAll: true,
     };
   }
 
+  // eslint-disable-next-line no-unused-vars
+  componentDidUpdate(prevProps, prevState, snapshot) {
```
[the remark this answers is not in the document]
Comment (humanumbrella) [reply src/components/ScriptExplorer/UTXOSet.jsx:46]: We don't - editor complains if those are missing, linter complains if they're there and unused. Removing

**Comment by humanumbrella** [comment]
I updated things to make sure that it at least shows that the address is now **selected** (full checkbox, not indeterminate or missing) when you navigate away and/or click edit transaction.

It will require more effort given the current design to actually save which inputs were selected. It would be easier with a larger refactor of how the application is structured.

**Comment by bucko13** [comment]
Don't know if this needs to be a blocker, but this was a weird little bug. If there's only one UTXO in an address, selecting it in the utxo selector doesn't work: 

https://user-images.githubusercontent.com/4344978/127078598-f1b7d0e5-9b92-4d59-afef-47ddbfb67fda.mp4

**Comment by bucko13** [comment]
Tested happy path and the reset of the utxos if going back to edit and it all seems to work other than the bug identified above. If that is easily fixable, then it'd be good if we can get that in (maybe just disabling that view if only one utxo as a quick fix? 🤷‍♂️ ). Otherwise it LGTM!

**Review by bucko13 — APPROVED** [review]
Confirmed it works! 👏 🎉 :shipit:

**Pull request merged**